cli/lsp/diagnostics.rs

```diff
@@ -203,15 +181,35 @@ impl DiagnosticsServer {
                   language_server.maybe_lint_config.clone(),
                 )
               };
-              update_diagnostics(
-                &client,
-                collection.clone(),
-                snapshot,
-                config,
-                maybe_lint_config,
-                &ts_server,
-                performance.clone(),
-              ).await;
+              let performance = performance.clone();
+              let ts_server = ts_server.clone();
+              let client = client.clone();
+              let token = token.clone();
+              let collection = collection.clone();
+              let previous_handle = handle.take();
+
+              handle = Some(tokio::spawn(async move {
+                if let Some(previous_handle) = previous_handle {
+                  // wait on the previous run to complete in order to prevent
+                  // multiple threads queueing up a lot of tsc requests
+                  tokio::select! {
+                    _ = token.cancelled() => { return; }
+                    _ = previous_handle => {}
+                  };
```

Comment:
This is an issue. It should always wait on the previous handle to complete and not cancel waiting.